### PR TITLE
Add kernel header tests to ci cd

### PR DESCRIPTION
**Pull request by samiura**

### CODE REVIEW

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -50,8 +57,11 @@ jobs:
       with:
         fetch-depth: 0
 
-    - name: build reducer
+    - name: build-and-reducer
```
Comment (jmw51798) [inline .github/workflows/build-and-test.yaml]: build-reducer?

File: .github/workflows/build-and-test.yaml [hunk 2]
```diff
@@ -50,8 +57,11 @@ jobs:
...
+      env:
+        PASS: ${{ secrets.DOCKER_PASSWORD }}
       run: |
+        echo $BENV_IMAGE
```
Comment (jmw51798) [inline .github/workflows/build-and-test.yaml]: remove?

File: .github/workflows/build-and-test.yaml [hunk 3]
```diff
@@ -200,3 +210,68 @@ jobs:
...
+        podman info
+        git submodule update  --init --recursive
+        
+        podman run -t --rm \
+        --mount type=bind,source=$PWD,destination=/root/src,readonly \
+        --mount type=bind,source=/var/run/docker.sock,destination=/var/run/docker.sock \
+        --env EBPF_NET_SRC_ROOT=/root/src \
+        --name benv \
+        --privileged \
+        quay.io/splunko11ytest/network-explorer-debug/build-env:latest \
```
Comment (jmw51798) [inline .github/workflows/build-and-test.yaml]: should this use $BENV_IMAGE?

File: test/kernel-headers/source/Vagrantfile [hunk 4]
```diff
@@ -1,3 +1,7 @@
+#!/bin/bash
```
Comment (jmw51798) [inline test/kernel-headers/source/Vagrantfile:1]: not a bash script - remove

File: test/kernel-headers/source/runners/0-setup.sh [hunk 5]
```diff
@@ -16,35 +19,45 @@ then
...
-# Push docker images to local docker registry
-docker tag kernel-collector localhost:5000/kernel-collector:latest
-docker push localhost:5000/kernel-collector
-docker tag reducer localhost:5000/reducer:latest
-docker push localhost:5000/reducer
-
 vagrant up --no-provision
 
-# Confirm that the vagrant VM is running
+# Confirm tha the vagrant VM is running
```
Comment (jmw51798) [inline test/kernel-headers/source/runners/0-setup.sh]: typo

File: test/kernel-headers/run-tests.sh [hunk 6]
```diff
@@ -32,6 +39,13 @@ do
   fi
 done
 
-set +x
```
Comment (jmw51798) [inline test/kernel-headers/run-tests.sh:35]: Leave this in

File: test/kernel-headers/run-tests.sh [hunk 7]
```diff
@@ -32,6 +39,13 @@ do
...
 echo -e "Ran ${num_run} tests, with ${num_failed} failure(s).\n" | tee -a ${test_dir}/summary.log
-echo "Test results are in ${test_dir}"
```
Comment (jmw51798) [inline test/kernel-headers/run-tests.sh]: Leave this change in

File: test/kernel-headers/source/runners/0-setup.sh [hunk 8]
```diff
@@ -16,35 +19,45 @@ then
...
 source "${EBPF_NET_SRC_ROOT}/dev/script/bash-error-lib.sh"
 
-# Check if local docker registry is running
-result=$(docker ps | grep registry:latest | grep local-docker-registry) || true
-if [[ "${result}" == "" ]]
-then
-  # local registry isn't running, so start it
-  "${EBPF_NET_SRC_ROOT}/dev/docker-registry.sh"
-fi
+vagrant plugin install vagrant-sshfs
```
Comment (jmw51798) [inline test/kernel-headers/source/runners/0-setup.sh]: Could this be done in the runner setup, not each time this script is run?

File: test/kernel-headers/source/runners/0-setup.sh [hunk 9]
```diff
@@ -16,35 +19,45 @@ then
...
 
-# Check that required docker images exist
-result1=$(docker images | grep ^kernel-collector) || true
-result2=$(docker images | grep ^reducer) || true
-if [[ "${result1}" == "" || "${result2}" == "" ]]
+vagrant plugin install vagrant-scp
```
Comment (jmw51798) [inline test/kernel-headers/source/runners/0-setup.sh]: Same - could this be done in the runner setup, not each time this script is run?

File: .github/workflows/build-and-test.yaml (hunk 3 above)
Comment (samiura) [reply .github/workflows/build-and-test.yaml]: yes

File: .github/workflows/build-and-test.yaml (hunk 2 above)
Comment (samiura) [reply .github/workflows/build-and-test.yaml]: yes

File: .github/workflows/build-and-test.yaml (hunk 1 above)
Comment (samiura) [reply .github/workflows/build-and-test.yaml]: yes

File: test/kernel-headers/source/Vagrantfile (hunk 4 above)
Comment (samiura) [reply test/kernel-headers/source/Vagrantfile:1]: yes

File: test/kernel-headers/source/runners/0-setup.sh (hunk 5 above)
Comment (samiura) [reply test/kernel-headers/source/runners/0-setup.sh]: yes

File: test/kernel-headers/run-tests.sh (hunk 6 above)
Comment (samiura) [reply test/kernel-headers/run-tests.sh:35]: yes

File: test/kernel-headers/run-tests.sh (hunk 7 above)
Comment (samiura) [reply test/kernel-headers/run-tests.sh]: yes

**Pull request closed without merging**